python/pyspark/rdd.py

```diff
@@ -791,9 +792,11 @@ def foreach(self, f):
         >>> def f(x): print(x)
         >>> sc.parallelize([1, 2, 3, 4, 5]).foreach(f)
         """
+        safe_f = fail_on_StopIteration(f)
```

Comment:
`safe` prefix doesn't imply why it's safe though .. I would just name it like `fail_on_stopiteration_f` or feel free to another name if you have a good one.

Reply:
Im okay with `safe` as is too if you feel strongly.